### PR TITLE
Removed <Report ID> from thank you page. Replaced with conditional Report Id display.

### LINKED ISSUE #1472: <ReportID> is displayed on thank you page
## Summary

<ReportID> is displayed on thank you page

error

## Steps to reproduce

> How exactly can the bug be reproduced? Be very specific.

## Unresolved questions

> Are there any related issues you consider out of scope for this issue that could be addressed in the future?

### PR DESCRIPTION
**Pull request by naomiseminega**
Fixes #1472

# Description
- Removed <Report ID> from thank you page
- Replaced with conditional Report Id display
- As per recommendations in Issue #1472:
    * When report ID starts with "NCFRS" msg is displayed that report submission was not successful
    * Otherwise , the generated report Id is displayed.
    *Also, figured we didn't need a line that says 'Submitting your report', since by the time there is a 'Thank You' page, the report has been submitted (it was misleading (?)).

# Required follow up work
- I noticed from the code that all report Id's that are generated start with NCFRS (?), not sure if this is a placeholder or will stay this way. We may have to update the logic in that case.

# Any new packages installed?
No
# Checklist:

- [x] I have looked at my code on GitHub and it all looks good (ex: no random commented out code or console.logs)
- [x] I have added and needed tests for my changes (in particular for new screens)
- [x] I have added a comment to any confusing code

### CODE REVIEW

File: f2/src/forms/__tests__/HowDidItStartForm.test.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import React from 'react'
```
Comment (sastels) [inline f2/src/forms/__tests__/HowDidItStartForm.test.js]: Isn't this test already in the code?

**Comment by sastels** [comment]
review app deployed at https://rac-prototyp-report-id-vaicqvy.herokuapp.com/

File: f2/src/ThankYouPage.js [hunk 2]
```diff
@@ -54,17 +54,23 @@ export const ThankYouPage = () => {
...
+                  </Text>
+                </Trans>
+              ) : (
+                <Trans
+                  id="thankYouPage.referenceNumber"
+                  values={{ reference: state.formData.reportId }}
+                >
+                  <Text as="span" d="block" />
+                </Trans>
+              )}
```
Comment (sastels) [inline f2/src/ThankYouPage.js]: * We should flip the order of these two blocks (ie show the reportId when `state.formData.reportId && state.formData.reportId.startsWith('NCFRS-')`)
* also, we should have a third option... something like when the confirmation page loads (ie in `ConfirmationPage.js` after the `const { i18n } = useLingui()`)
```
if formdata.reportId !== '') 
 dispatch({ type: 'saveFormData', data: { reportId: '' } })
```
 to make the reportId an empty string. Then on the thankyou page we could also check for `reportId ===''` and if so show something like "Submitting report..." (which would then get replaced by the number or the error message when there's a response from the server)

Comment (naomiseminega) [reply f2/src/ThankYouPage.js]: Sounds good! I'm on it

File: f2/src/forms/__tests__/HowDidItStartForm.test.js (hunk 1 above)
Comment (naomiseminega) [reply f2/src/forms/__tests__/HowDidItStartForm.test.js]: Yes it is! Oups, not sure how it ended up here again :P

File: f2/src/ThankYouPage.js (hunk 2 above)
Comment (naomiseminega) [reply f2/src/ThankYouPage.js]: @sastels fixed! Hopefully I covered all the cases :)

File: f2/src/ThankYouPage.js [hunk 3]
```diff
@@ -29,6 +29,22 @@ export const ThankYouPage = () => {
 
   const [, dispatch] = useStateValue()
 
+  // Message displayed on Thank you Page
+  let reportId = state.formData.reportId
+  let thankYouMessage
+
+  if (!reportId || reportId === '') {
+    thankYouMessage = <Trans id="thankYouPage.reportSubmission" />
```
Comment (sastels) [inline f2/src/ThankYouPage.js]: I didn't know you could do this. Nice! 🎉

Comment (naomiseminega) [reply f2/src/ThankYouPage.js]: Me either! Lots of googling :p :p

**Pull request merged**